test/test_katello/test_enabled_report.py

```diff
@@ -19,3 +19,12 @@ def test_valid(self):
                                'repositoryid': 'enabled'}]}}
 
         self.assertEqual(expected, report.content)
+
+    @patch('katello.enabled_report.EnabledReport._replace_vars')
+    def test_var_interpolation(self, mock_replace_vars):
```

Comment:
@beav yeah, getting the testing just the way you want across these OS/Python versions can be a pain. The way this test is written circumvents  _replace_vars and _obtain_mappings (I think you were pointing that out) so something should be changed so we can assert against real output from `_obtain_mappings`

But ... I'm having a hard time figuring out how to properly mock out the yum/dnf base so we can set our own vars to assert against. Maybe a compromise is to duplicate some of the code from your impl in this test that way we can get an end-to-end test of the report output? Or hard-code a mapping of a few os => releasevar (assume x86_64 basearch) so all the docker testing will pass? Let me know what you think!

Reply:
@jturel I think the best bet would be for me to split the dnf and yum detection code into two methods, and then mock those. That would at least give the ability to test the `ImportError` fall-back from one to the other.

I'm not sure of the best way to automate a "real world" test of this across dists. We may need to create a small server listening on localhost or something that receives repo lists and validates it, and that can run across 5+6+7+F28+zypper.

For this PR, would it be OK to just make it easier to stub the os-specific code and test to that?